filechooser-portal/Main.vala

```diff
@@ -343,10 +332,61 @@ public class Files.FileChooserPortal : Object {
         yield;
 
         dialogs.remove (parent_window);
+        dialog.destroy ();
+
         response = _response;
         results = _results;
     }
 
+    private async bool check_overwrite_uri (string? uri) {
+        if (uri == null) {
+            return false;
+        }
+
+        var file = GLib.File.new_for_uri (uri);
+        if (file.query_exists ()) {
+            var display_name = file.get_basename ();
+            var primary = _("Replace “%s”?").printf (display_name);
+            unowned var secondary = _("Replacing this file will overwrite its current contents");
+            if (file.query_file_type (FileQueryInfoFlags.NOFOLLOW_SYMLINKS) == FileType.SYMBOLIC_LINK) {
+                try {
+                    var info = file.query_info (FileAttribute.STANDARD_SYMLINK_TARGET, FileQueryInfoFlags.NONE);
+                    if (info != null) {
+                        primary = _("Replace “%s”?".printf (info.get_symlink_target ()));
+                    } else {
+                        primary = _("Replace the target of “%s”?".printf (display_name));
+                    }
+
+                    secondary = _("Replacing the target file for this link will overwrite its current contents.");
+                } catch (Error e) {
+                    warning ("Could not get info for %s", file.get_uri ());
+                }
+            }
+
+            var replace_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+                primary, secondary, "dialog-warning", Gtk.ButtonsType.CANCEL
+            );
+
+            replace_dialog.set_modal (true);
+
+            var replace_button = replace_dialog.add_button ("Replace", Gtk.ResponseType.YES);
+            replace_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
+
+            var response_ok = false;
+            replace_dialog.response.connect ((id) => {
+                response_ok = id == Gtk.ResponseType.YES;
+                check_overwrite_uri.callback ();
+            });
+
+            replace_dialog.show_all ();
```

Comment:
```suggestion
            replace_dialog.present ();
```

`Gtk.Widget.show_all()` was removed in Gtk4 too, since non-toplevels Widgets are shown by default, Granite.MessageDialog already make they childs visible in Gtk3, so calling `Gtk.Window.present()` works here.